core/chain_manager.go

```diff
@@ -487,15 +546,37 @@ const (
 	SideStatTy
 )
 
+/*
```

Comment:
Ah, this will definitely be needed when implementing the fast sync algo, however, since it's hard to design the needed mechanisms exactly, I commented it out so it doesn't become part of the API too soon. But it's definitely going in with potentially some modifications. If it's bothersome, I could remove it.